app/models/qdm/tacoma/valueset.rb

```diff
@@ -0,0 +1,68 @@
+module QDM
+  # ValueSet represents a collection of Concepts, used by the Measures to specify a set of codes for a particular topic
+  class ValueSet
+    include Mongoid::Document
+    field :oid, type: String
+    field :display_name, type: String
+    field :version, type: String
+    field :user_id, type: String # Eventually we need to delete this from bundles when exporting
+    field :categories, type: Hash
+
+    index(oid: 1)
+    embeds_many :concepts
+    index 'concepts.code' => 1
+    index 'concepts.code_system' => 1
+    index 'concepts.code_system_name' => 1
+    index 'bundle_id' => 1
+    scope :by_oid, ->(oid) { where(oid: oid) }
+
+    # Provides an Array of Hashes. Each code system gets its own Hash
+    # The hash has a key of "set" for the code system name and "values"
+    # for the actual code list
+    def code_set_map
+      codes = []
+      concept_pairs = concepts.each_with_object({}) do |concept, memo|
+        memo[concept.code_system_name] ||= []
+        memo[concept.code_system_name] << concept.code
+      end
+      concept_pairs.each_pair do |code_set, code_list|
+        codes << { 'set' => code_set, 'values' => code_list }
+      end
+
+      codes
+    end
+
+    def self.load_from_xml(doc)
+      doc.root.add_namespace_definition('vs', 'urn:ihe:iti:svs:2008')
```

Comment:
Was this code extracted from HDS? Would it be more appropriate for value set parsing code to be put elsewhere?

Reply:
It was extracted from HDS. Does the Bonnie team have a location preference for it? I think it's only used by Bonnie; cypress never has to deal with loading from XML.